Dockerfile

```diff
@@ -2,20 +2,37 @@ FROM ubuntu:trusty
 MAINTAINER Akeem McLennon <akeem@mclennon.com>
 
 USER root
-RUN apt-get -y update
-RUN apt-get -y dist-upgrade
-RUN apt-get -y build-dep phantomjs
-RUN apt-get install -y python flex bison gperf ruby perl \
-  libsqlite3-dev libfontconfig1-dev libicu-dev libfreetype6 libssl-dev \
-  libpng-dev libjpeg-dev
-RUN apt-get -y install git-core
-WORKDIR /usr/local
-RUN git clone https://github.com/AkeemMcLennon/phantomjs.git
+
+RUN mkdir /usr/local/phantomjs
+
 WORKDIR /usr/local/phantomjs
-RUN ./build.sh --confirm
-RUN ln -s /usr/local/phantomjs/bin/phantomjs /usr/bin/phantomjs 
-RUN apt-get autoremove -y
-RUN apt-get clean all
-ADD join-hub.sh /usr/local/phantomjs/
-ENTRYPOINT sh /usr/local/phantomjs/join-hub.sh
 
+ARG CLEAN=0
+
+RUN apt-get -y update \
+      && apt-get -y dist-upgrade \
+      && apt-get -y build-dep phantomjs \
+      && apt-get install -y python flex bison gperf ruby perl \
+                            libsqlite3-dev libfontconfig1-dev \
+                            libicu-dev libfreetype6 libssl-dev \
+                            libpng-dev libjpeg-dev \
+                            git-core libicu52 \
+                            curl \
+      && curl -L https://github.com/AkeemMcLennon/phantomjs/archive/2.0.0.tar.gz | tar --strip-components=1 -xzf - \
```

Comment:
The problem is, that each RUN command will create an intermediate image. If this is bigger than the next one (because the next command is a `apt-get clean` or stuff like this) then this next image won't shrink the image afterwards. So doing this in multiple RUN commands means, the resulting image will have huge intermediate images you have to download as well. Thats what I want to avoid (so not downloading 2GB but only ~450MB).